tests/test_domains/test_domain_cpp.py

```diff
@@ -1099,77 +1109,85 @@ def parse_template_parameter(param: str):
 #         print(a)
 #     raise DefinitionError
 
+E2E_TESTROOT_ID = 'domain-cpp'
+sphinx_domain_cpp = stack_pytest_markers(  # test with 'html' build to some files written
+    pytest.mark.sphinx('html', testroot=E2E_TESTROOT_ID, confoverrides={'nitpicky': True}),
+    pytest.mark.test_params(shared_result=f'{E2E_TESTROOT_ID}-{uuid.uuid4().hex}'),
+)
+
 
-def filter_warnings(warning, file):
-    lines = warning.getvalue().split("\n")
-    res = [l for l in lines if "domain-cpp" in l and f"{file}.rst" in l and
-           "WARNING: document isn't included in any toctree" not in l]
+def filter_warnings(warning, file, testroot_id=E2E_TESTROOT_ID):
+    lines = strip_escseq(warning.getvalue()).split("\n")
+    res = [l for l in lines if testroot_id in l and f"{file}.rst" in l
+           and "WARNING: document isn't included in any toctree" not in l]
     print(f"Filtered warnings for file '{file}':")
     for w in res:
         print(w)
     return res
 
 
-@pytest.mark.sphinx(testroot='domain-cpp', confoverrides={'nitpicky': True})
+@sphinx_domain_cpp
 def test_domain_cpp_build_multi_decl_lookup(app, status, warning):
-    app.build(force_all=True)
+    app.build()
     ws = filter_warnings(warning, "lookup-key-overload")
     assert len(ws) == 0
 
     ws = filter_warnings(warning, "multi-decl-lookup")
     assert len(ws) == 0
 
 
-@pytest.mark.sphinx(testroot='domain-cpp', confoverrides={'nitpicky': True})
+@sphinx_domain_cpp
 def test_domain_cpp_build_warn_template_param_qualified_name(app, status, warning):
-    app.build(force_all=True)
+    app.build()
     ws = filter_warnings(warning, "warn-template-param-qualified-name")
     assert len(ws) == 2
     assert "WARNING: cpp:type reference target not found: T::typeWarn" in ws[0]
     assert "WARNING: cpp:type reference target not found: T::U::typeWarn" in ws[1]
 
 
-@pytest.mark.sphinx(testroot='domain-cpp', confoverrides={'nitpicky': True})
+@sphinx_domain_cpp
 def test_domain_cpp_build_backslash_ok_true(app, status, warning):
-    app.build(force_all=True)
+    app.build()
     ws = filter_warnings(warning, "backslash")
     assert len(ws) == 0
 
 
-@pytest.mark.sphinx(testroot='domain-cpp', confoverrides={'nitpicky': True})
+@sphinx_domain_cpp
 def test_domain_cpp_build_semicolon(app, status, warning):
-    app.build(force_all=True)
+    app.build()
     ws = filter_warnings(warning, "semicolon")
-    assert len(ws) == 0
+    assert len(ws) == 1
```

Comment:
This one was *not* expected because of the execution order but if you look at the source files you'll see that there is a duplicated definition.